Lab2/Assign2/main.cpp

```diff
@@ -0,0 +1,36 @@
+//
+//  main.cpp
+//  Assign2
+//
+//  Created by Monisha Gunalan2 on 2017-03-03.
+//  Copyright © 2017 Monisha Gunalan2. All rights reserved.
+//
+
+#include <iostream>
+using namespace std;
+#include "Student.hpp"
+#include "Teacher.hpp"
+#include "Course.hpp"
+#include <vector>
+
+int main(int argc, const char * argv[]) {
+
+    
+    Student *student1 = new Student("John", "Doe", 20, "123 main street", "ottawa", "613 123 1234");
+    Student *student2 = new Student("James", "Doe", 20, "1 first ave", "ottawa", "613 888 8888");
+    Student *student3 = new Student("Kevin", "Doe", 20, "4 march road", "ottawa", "613 111 1111");
+    
+    Teacher *teacher = new Teacher("David", "Walter", 50, "123 main street", "ottawa", "613 123 1234");
+    
+    Course *course = new Course("C++");
```

Comment:
Likewise for course and teacher, they should not be dynamically allocated.

Remember every `new` must be paired with a `delete`. In real world code not `delete`ing would lead to a memory leak. Also, instantiating on the stack is faster because the memory is already available to you and doesn't need to be requested from the OS. The best part about it is that it cleans up after itself as soon as it leaves the scope.

In general, prefer 'automatic allocation' (colloquially "stack" or scope-based allocation) to dynamic allocation except for when the lifetime of the object needs to extend beyond the scope or for really large objects. (And even then you will get into smart pointers such as std::shared_ptr and std::unique_ptr (or our equivalents CYISharedPtr, CYIScopedPtr) and generally avoid "naked `new`".) Automatic allocation is actually the basis of a very important concept in writing good C++ called RAII.